### DISCUSSION TITLE
How to Update Child's Relations Using Create API

### DISCUSSION THREAD

**Discussion by mithatercan** [prisma/prisma#16507] [Q&A] [answered]
I've included an example of my code, which includes three models. I want to update the customer's device id when I create a new sale, but I can't accomplish it using the connect or update key.

``` prisma

model Device {
  id        String        @id @default(uuid())
  sales     Sale[]        @relation("SaleDevice")
  customers Customer[]    @relation("CustomerDevice")
}

model Sale {
  id            String            @id @default(uuid())
  device        Device?           @relation("SaleDevice", fields: [deviceId], references: [id], onDelete: Cascade)
  deviceId      String?
  customer      Customer          @relation("CustomerSale", fields: [customerId], references: [id], onDelete: Cascade)
  customerId    String
  products      CustomerProduct[] @relation("SaleCustomerProduct")
}


model Customer {
  id        String            @id @default(uuid())
  purchases Sale[]            @relation("CustomerSale")
  device    Device?           @relation("CustomerDevice", fields: [deviceId], references: [id], onDelete: SetNull)
  deviceId  String?
}

```

A new sale creation example

```javascript
const sale = await prisma.sale.create({
  data: {
    device: {
      connect: {
        id: deviceId,
      },
    },
    customer: {
      connect: {
        id: customerId,
        device: {
          // This is what I'm attempting to achieve but failing.
          connect: {
            id: deviceId,
          },
        },
      },
    },
  },
})

```

**Comment by pantharshit00** [comment, marked as answer]
Hello @mithatercan,

You can only `connect` or `create` a nested entity when you are creating a record. You will need to use a separate call here to update the customer.

You can also use [interactive transactions ](https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide#interactive-transactions) here if you want these operations to run in a single transaction.

**Reply by mithatercan** [reply]
Hello, @pantharshit00.

Thank you for your time in responding. Yes, I used transactions in this case.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
